### ISSUE TITLE
Feature request - store split pane state in local storage with provided key

### ISSUE THREAD

**Issue by BlindDespair** [bertrandg/angular-split#160]
It would be great to be able to remember the state of split panes in LocalStorage so that next time when app is reopened it would be in the same state. We're currently using https://github.com/wannabegeek/ng2-split-pane in our project and it has that feature it's pretty convenient. But that module is not performance friendly and does not have option to have multiple panes inside split, so guys did some nasty workarounds having a splitpane inside one side of the other splitpane which I personally don't really like because it's not working correctly. So now I am looking towards this module, which looks very promising so far.

In short:
Please, add input to split container [localStorageKey], put a state of pane to LocalStorage at the end of each resizing. When split container is initialized load its state from LocalStorage and validate it whether it fits together with what is being initialized, then use it as starting point.

**Comment by bertrandg** [comment]
hi @BlindDespair 
It's a bit out of library scope but why not...
But it will not be soon (no time currently) so you can do it yourself easily like demo app example ;)
https://bertrandg.github.io/angular-split/#/examples/workspace-localstorage
https://github.com/bertrandg/angular-split/blob/master/src_app/app/component/examples/workspaceLocalstorage.route.component.ts

**Comment by BlindDespair** [comment]
Thank you for fast response, that actually looks good enough and with right abstraction can be considered as a solution to go. I will suggest it to my team.

**Comment by beeman** [comment]
I agree with @bertrandg that this is a bit out of scope, and it's easy to implement following the demo.

Considering @BlindDespair is happy with the solution, I'm closing this issue.
